func_test.go

```diff
@@ -148,12 +113,9 @@ func TestFuncWrongRet(t *testing.T) {
 		defer func() { caught = recover() }()
 		f.Call(store)
 	}()
-	if caught == nil {
-		panic("expected a panic")
-	}
-	if !strings.Contains(caught.(string), "callback produced wrong type of result") {
-		panic(fmt.Sprintf("wrong panic message %s", caught))
-	}
+	assert.NotNil(t, caught, "expected a panic")
+	assert.IsType(t, caught, string(""))
+	assert.Containsf(t, caught.(string), "callback produced wrong type of result", "wrong panic message %s", caught)
```

Comment:
I think this could still panic if the condition isn't met. `assert.*` will report an error but not abort the test execution. The used package has `require.*` variants that will call t.Fail if the condition isn't met.

Reply:
Exactly!